src/utils/equipment.ts

```diff
@@ -0,0 +1,90 @@
+import Equipment from "../resources/Equipment";
+// Filtering Function to reduce the size of the equipment array being passed down
+export function queryEquipment(
+  equipment: Equipment[],
+  searchString: string
+): Equipment[] {
+  let queriedEquipment: Equipment[] = [];
+  if (searchString !== "") {
+    const queries = searchString.split(/\W+/);
+    queriedEquipment = equipment.filter(function (equipment) {
+      return queries.every(function (query) {
+        return (
+          equipment.description
+            .toLowerCase()
+            .includes(query.toLowerCase().trim()) ||
+          equipment.category
+            .toLowerCase()
+            .includes(query.toLowerCase().trim()) ||
+          equipment.tags.some(function (tag) {
+            return tag.name.toLowerCase().includes(query.toLowerCase().trim());
+          })
+        );
+      });
+    });
+  } else {
+    queriedEquipment = equipment;
+  }
+  return queriedEquipment;
+}
+
+// Filtering Function to reduce the size of the equipment array being passed down
+export function filterEquipment(
+  equipment: Equipment[],
+  filters: { [k: string]: boolean }
+): Equipment[] | undefined {
+  const activeFilters = Object.keys(filters).filter(function (key: string) {
+    return filters[key];
+  });
+  return equipment.filter(function (item) {
+    return activeFilters.every(function (filter) {
+      return item.tags.some(function (tag) {
+        return tag.name.toLowerCase().includes(filter.toLowerCase().trim());
+      });
+    });
+  });
+}
+
+// Function to convert equipment Array to Quantized equipment Array
+export function quantizeEquipment(equipment: Equipment[]): Equipment[] {
+  const tempArray: Equipment[] = [];
+  equipment.forEach((item) => {
+    item.quantity = equipment.filter(
+      (element) => element.description === item.description
+    ).length;
+    const index = tempArray.findIndex(
+      (element) => element.description === item.description
+    );
+    if (index === -1) {
+      tempArray.push(item);
+    }
+  });
+  return tempArray;
+}
```

Comment:
Moving my comment from the old file to the new file...
* You're mutating the original equipment array, reassigning `item.quantity`.  Not good!  Copy the item into the new array (use `Object.assign({}, item)` or the shorthand `({...item})` to copy an object).
* `temp` has no semantic meaning, and moreover you are returning this supposed "temp" array.  Call it what it is: the quantized array.
* if you don't care about the index, use `find` rather than `findIndex`
```suggestion
// Function to convert equipment Array to Quantized equipment Array
export function quantizeEquipment(gear: Equipment[]): Equipment[] {
  return gear.reduce((quantized, item) => {
    const hasTitle = (gear) => gear.title === item.title;
    const alreadyDone = quantized.find(hasTitle);
    if (!alreadyDone)
      quantized.push({
        ...item,
        quantity: gear.filter(hasTitle).length,
      });
    return quantized;
  }, []);
}
```